src/frontend/context/SecurityContext.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+import { PASSWORD_KEY } from "../constants";
+
+type AppModeTypes = "normal" | "kill";
+type AuthStatusTypes = "pending" | "authenticated" | "notRequired";
+
+type AuthState = {
+  passcode: string | null;
+  appMode: AppModeTypes;
+  killModeEnabled: boolean;
+  authStatus: AuthStatusTypes;
+};
+
+type KillModeActions =
+  | { type: "setPasscode"; newPasscode: string | null }
+  | { type: "toggleAppMode"; newAppMode?: AppModeTypes }
+  | { type: "toggleKillModeEnabled"; newKillModeValue?: boolean }
+  | { type: "setAuthStatus"; newAuthStatus: AuthStatusTypes };
+
+function killModeReducer(state: AuthState, action: KillModeActions): AuthState {
```

Comment:
would rename to something more general and related to Security since it's not just handling kill mode state